connectors/source.py

```diff
@@ -107,6 +132,114 @@ def is_empty(self):
     def to_dict(self):
         return dict(self._raw_config)
 
+    def is_valid(self):
+        """Validates every Field against its `validations`.
+
+        Raises ConfigurableFieldValueError if any validation errors are found,
+        otherwise returns True.
+        """
+        validation_errors = []
+
+        for _, field in self._config.items():
+            if not self.dependencies_satisfied(field):
+                # we don't validate a field if its dependencies are not met
+                logger.info(
+                    f"Field {field.label} was not validated because its dependencies were not met."
+                )
+                continue
+
+            validation_errors = validation_errors + self.validate_field(field)
+
+        if len(validation_errors) > 0:
+            raise ConfigurableFieldValueError(
+                f"Field validation errors: {'; '.join(validation_errors)}"
+            )
+
+        return True
+
+    def dependencies_satisfied(self, field):
+        """Used to check if a Field has its dependencies satisfied.
+
+        Returns True if all dependencies are satisfied, or no dependencies exist.
+        Returns False if one or more dependencies are not satisfied.
+        """
+        if len(field.depends_on) <= 0:
+            return True
+
+        satisfied = True
+
+        for dependency in field.depends_on:
+            if dependency["field"] not in self._config:
+                # cannot check dependency if field does not exist
+                raise ConfigurableFieldDependencyError(
+                    f'Configuration `{field.label}` depends on configuration `{dependency["field"]}`, but it does not exist.'
+                )
+
+            if self._config[dependency["field"]].value != dependency["value"]:
+                satisfied = False
```

Comment:
You can probably get ridd of the `satisfied` variable and return `False` here since the first occurrence is enough to return